packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -40,14 +38,39 @@ type ShouldSummarizeState =
     | { shouldSummarize: true; }
     | { shouldSummarize: false; stopReason: StopReason; };
 
-export class SummaryManager extends EventEmitter implements IDisposable {
+export interface IConnectedEvents extends IEvent {
```

Comment:
is this worth using elsewhere? this pair of events seems pretty common...

Reply:
Yes definitely :)  though I haven't generalized yet.  I think we can do similar dependency injection patterns in some other places (maybe on the `setConnectionState` chain for example) and then standardize across the usages.